### PR TITLE
[pulsar-broker] Fix: Topic dedup policy configuration doesn't support V1 topic-name

### PR DESCRIPTION
**Pull request by rdhabalia**
### Motivation
Right now, Topic level dedup policy configuration doesn't support V1 topic with cluster name in it.

**NOTE:** This is PR is on top of: #8555

### CODE REVIEW

**Review by eolivelli — APPROVED** [review]
LGTM

**Comment by zymap** [comment]
Record failed tests and rerun it.

```
Error:  Failures: 
Error:  org.apache.pulsar.broker.admin.AdminApiTest2.testIncrementPartitionsOfTopic(org.apache.pulsar.broker.admin.AdminApiTest2)
[INFO]   Run 1: PASS
Error:    Run 2: AdminApiTest2.testIncrementPartitionsOfTopic:215 expected [8] but found [9]
[INFO] 
Error:  org.apache.pulsar.broker.admin.AdminApiTest2.testNonPersistentTopics(org.apache.pulsar.broker.admin.AdminApiTest2)
[INFO]   Run 1: PASS
Error:    Run 2: AdminApiTest2.testNonPersistentTopics:756 expected [100] but found [101]
[INFO] 
Error:  org.apache.pulsar.broker.admin.TopicPoliciesTest.testPublishRateInDifferentLevelPolicy(org.apache.pulsar.broker.admin.TopicPoliciesTest)
[INFO]   Run 1: PASS
Error:    Run 2: TopicPoliciesTest.testPublishRateInDifferentLevelPolicy:1196 expected [10] but found [11]
[INFO] 
```

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/PersistentTopics.java [hunk 1]
```diff
@@ -347,6 +348,67 @@ public void getPartitionedStats(@Suspended final AsyncResponse asyncResponse,
...
+    @ApiResponses(value = { @ApiResponse(code = 403, message = "Don't have admin permission"),
+            @ApiResponse(code = 404, message = "Tenant or cluster or namespace or topic doesn't exist"),
+            @ApiResponse(code = 405, message = "Topic level policy is disabled, to enable the topic level policy and retry")})
+    public void getDeduplicationEnabled(@Suspended final AsyncResponse asyncResponse,
+                             @PathParam("tenant") String tenant,
+                             @PathParam("cluster") String cluster,
+                             @PathParam("namespace") String namespace,
+                             @PathParam("topic") @Encoded String encodedTopic) {
+        validateTopicName(tenant, cluster, namespace, encodedTopic);
+        TopicPolicies topicPolicies = getTopicPolicies(topicName).orElse(new TopicPolicies());
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/PersistentTopics.java:363]: Should we check the topic ownership check?

**Comment by codelipenghui** [comment]
@rdhabalia Seems dup with #8555 ?

**Comment by rdhabalia** [comment]
@codelipenghui #8555 is to add rest-api but this PR actually fix dedup topic policy logic which doesn't support v1-topic name with `cluster-name` in it.

**Review by codelipenghui — APPROVED** [review]

**Comment by zymap** [comment]
Because this PR is opened for a long time. I will close this PR. Feel free to reopen it if you needed.

**Pull request closed without merging**